packages/taiko-client/driver/soft_blocks/api.go

```diff
@@ -0,0 +1,112 @@
+package preconfserver
+
+import (
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/labstack/echo/v4"
+)
+
+// TransactionBatchMarker represents the status of a preconfirmation transactions group.
+type TransactionBatchMarker string
+
+// PreconfBlockBatchMarker values.
+const (
+	BatchMarkerEOB TransactionBatchMarker = "end_of_block"
+	BatchMarkerEOP TransactionBatchMarker = "end_of_preconf"
+)
+
+// TransactionBatch represents a preconfirmation block group.
+type TransactionBatch struct {
+	BlockID          uint64                 `json:"blockId"`
+	ID               uint64                 `json:"batchId"`
+	TransactionsList types.Transactions     `json:"transactions"`
+	BatchMarker      TransactionBatchMarker `json:"batchType"`
+	Signature        string                 `json:"signature"`
+
+	// Block parameters
+	Timestamp             uint64         `json:"timestamp"`
```

Comment:
we only need to ask for values that the proposer can decide, not those our protocol contracts will calculate:

- params.timestamp
- params.anchorBlockId
- params.coinbase
- anchorStateRoot
- anchorBlockHash

It will be great if the server can query the right `anchorStateRoot`, and `anchorBlockHash` values to prevent a malicious preconfer to use the wrong values -- image the wrong values are used and Taikoscan display total bad blocks with these incorrect values.